client/src/partials/primary_cash/income/generic/generic.js

```diff
@@ -1,232 +1,218 @@
+    }
+  };
+
+  // Expose to view
+  $scope.update           = update;
+  $scope.setCurrency      = setCurrency;
+  $scope.formatAccount    = formatAccount;
+  $scope.formatDebtor     = formatDebtor;
+  $scope.generate         = generate;
+  $scope.clear            = clear;
+  $scope.submit           = submit;
+  $scope.reconfigure      = reconfigure;
+  $scope.setConfiguration = setConfiguration;
+
+  // Watchers
+  $scope.$watch('session.receipt', valid, true);
+  $scope.$watch('session.currency', valid, true);
+
+  // Startup
+  startup();
+  cache.fetch('currency').then(load);
+  cache.fetch('account').then(getAccount);
+
+  // Functions
+  function load (currency) {
+    if (!currency) { return; }
+     $scope.session.currency = currency;
+  }
+
+  function getAccount (ac) {
+    if (!ac) { return; }
+     session.configured = true;
+     session.ac = ac;
+     session.complete = true;
+  }
+
+  function startup() {
+    if (!exchange.hasDailyRate()) { $location.path('/primary_cash/'); }
+
     if (Number.isNaN(Number($routeParams.id))) {
       throw new Error('No cashbox selected');
     }
 
     isDefined = angular.isDefined;
-
     $scope.timestamp = new Date();
 
     // session.today = $scope.timestamp.toISOString().slice(0, 10);
-    //session.today = util.htmlDate(new Date());
     tomorrow = new Date();
     tomorrow.setDate(tomorrow.getDate() + 1);
     session.tomorrow = util.htmlDate(tomorrow);
 
-    dependencies.currencies = {
-      query : {
-        tables : {
-          'currency' : {
-            columns : ['id', 'name', 'symbol']
-          }
-        }
-      }
-    };
-
-    dependencies.projects = {
-      query : {
-        tables : {
-          'project' : {
-            columns : ['id', 'abbr']
-          }
-        }
-      }
-    };
-
-    dependencies.accounts = {
-      query : {
-        tables : {
-          'account' :{
-            columns : ['id', 'account_txt', 'account_number', 'is_ohada']
-          }
-        },
-        where : ['account.is_ohada=1', 'AND', 'account.account_type_id<>3']
-      }
-    };
+    // init models
+    $scope.project =  SessionService.project;
+    dependencies.projects.query.where = ['project.enterprise_id=' + $scope.project.enterprise_id];
+    validate.process(dependencies)
+    .then(function (models) {
+      angular.extend($scope, models);
+      session.receipt.date = new Date();
+      session.receipt.cost = 0.00;
+      session.receipt.cash_box_id = $routeParams.id;
+      session.accounts = models.accounts.data;
+    })
+    .catch(error);
+  }
 
-    cache.fetch('currency').then(load);
-    cache.fetch('account').then(getAccount);
+  function formatDebtor(debtor) {
+    return [
+      '[' + $scope.projects.get(debtor.project_id).abbr,
+      debtor.reference + ']',
+      debtor.first_name,
+      debtor.last_name
+    ].join(' ');
+  }
 
+  function generate () {
+    session.receipt.reference_uuid = uuid();
```

Comment:
I would remove this function.  It can be done within the submit portion of the code, or at startup.